test/ha/controller_test.go

```diff
@@ -49,19 +52,23 @@ func TestControllerHA(t *testing.T) {
 	test.CleanupOnInterrupt(func() { test.TearDown(clients, service1Names) })
 	defer test.TearDown(clients, service1Names)
 
-	clients.KubeClient.Kube.CoreV1().Pods(system.Namespace()).Delete(leaderController, &metav1.DeleteOptions{})
+	prober := test.RunRouteProber(t.Logf, clients, resources.Service.Status.URL.URL())
+	defer test.AssertProberDefault(t, prober)
```

Comment:
This seems to change semantics of the test rather than being just reuse?

Reply:
Yes. I'll add it to the description.

Reply:
Actually. I'll send it in a separate PR. The test failed. Sorry.